### DISCUSSION TITLE
selecting specific interface for zenoh communication

### DISCUSSION THREAD

**Discussion by ros2torial** [eclipse-zenoh/roadmap#83] [Q&A] [answered]
Hi,
I am running a zenoh client in a robot PC(192.168.8.24) which is having many active physical network ports(internet, lidar etc.) and a zenoh router in another PC(192.168.7.17). Both the PC's are on different LAN network layers and connected through an access point which is NAT enabled (192,168.8.1->192.168.7.250). If the robot PC is only connected access point and other connections like internet is removed then zenoh client is able to communicate with zenoh router. But, if other network ports are connected then I am unable to establish communication between zenoh client and router.  I found an option in scouting multicast to specify the interface which I guess is the network interface name we get by `ifconfig` command. In my case, it was interface **eth0** from which zenoh client was connected to zenoh router.

I added the below section in my zenoh client config file.

```
scouting: {
    timeout: 3000,
    delay: 200,
    multicast: {
      enabled: true,
      address: "224.0.0.224:7446",
      interface: "eth0", 
      autoconnect: { router: "", peer: "router|peer" },
      listen: true,
    },
    },
```

commands used - 
PC 1
`zenohd -l udp/192.168.7.17:7447`

PC 2 (robot PC)
`zenoh-bridge-dds -m client -e udp/192.168.7.17:7447`

Still zenoh communication is not established. Any suggestion, what is wrong here.
@JEnoch

**Comment by ros2torial** [comment]
@JEnoch is there any example to select specific interface to solve this issue.

**Reply by JEnoch** [reply, marked as answer]
The scouting protocol over UDP multicast is used for discovery when the zenoh entity to connect to is not specified in the startup configuration.
Here you configure the bridge to connect to the router at startup via UDP unicast. Thus scouting is not required in this case.

It could rather be a problem a NAT issue. It's possibly not able or not configured to automatically add IP translation back from 192.168.7.250:<random_port> to 192.168.8.24:<random_port> (where <random_port> is the UDP port open by the bridge and used to send messages).

Could you provide the logs of both bridge and router in the 2 cases (working with only connected access point, and not working with other network ports connected). Run them as such:
 - `RUST_LOG=zenoh_transport=trace zenohd -l udp/192.168.7.17:7447`
 - `RUST_LOG=zenoh_transport=trace zenoh-bridge-dds -m client -e udp/192.168.7.17:7447`
